### ISSUE TITLE
CORS issue while connecting with basic auth

### ISSUE THREAD

**Issue by farhan687** [elastic/elasticsearch#20905]
My goal is to access elasticsearch data from other domain, am I missing anything over here?

I set the config in elasticsearch.yml file which looks like this

```yml
http.cors.enabled: true
http.cors.allow-origin: "/.*/"
http.cors.allow-credentials: true
http.cors.allow-headers: "Authorization, X-Requested-With, Content-Type, Content-Length"
```

after set the config I restart elasticsearch

now I want to do xhr call from other domain
```js
var xhr = new XMLHttpRequest();
xhr.open('POST', 'http://127.0.0.1:9200/_search', true);
xhr.withCredentials = true;
xhr.setRequestHeader('Content-Type', 'application/json');
xhr.setRequestHeader('Authorization', 'Basic ' + btoa('admin:admin'));
xhr.send('{"query":{"match_all":{}}}');
```

It responds with following error

XMLHttpRequest cannot load http://127.0.0.1:9200/_search. Request header field Authorization is not allowed by Access-Control-Allow-Headers in preflight response.

**Comment by jasontedor** [comment]
Would you please provide the information that was requested in the issue template, especially the version of Elasticsearch?

**Comment by farhan687** [comment]
Hey @jasontedor 
It is Elasticsearch Version 2.3.

**Comment by jasontedor** [comment]
Duplicates #17483, fixed in 2.3.2.
